src/module_4/model_api.py

```diff
@@ -0,0 +1,215 @@
+import json
+import pandas as pd
+import os
+import pickle
+from sklearn.pipeline import Pipeline
+from typing import List, NamedTuple
+
+
+class ExpectedEventKeysFit(NamedTuple):
+    model_parametrisation: dict
+    train_cols: list[str]
+    target: str
+    data: any
+    pipe: str
+    pipeline: callable
+    date: str
+
+
+class ExpectedEventKeysPredict(NamedTuple):
+    date: str
+    users: str
+
+
+def perim_modelo(df: pd.DataFrame, min_items: int = 5) -> pd.DataFrame:
+    """
+    Given a df that has columns id | order_id it generates a new dataframe which
+    contains the orders that have more items than the argument min_items
+    """
+    assert {"variant_id", "order_id"}.issubset(set(df.columns))
```

Comment:
Is a good practice to embed asserts inside try catchs, so that you can capture and log the errors properly